plugins/org.eclipse.fordiac.ide.gef/src/org/eclipse/fordiac/ide/gef/DiagramEditor.java

```diff
@@ -276,13 +277,25 @@ protected void removeAnnotationModelDispatcher() {
 	 */
 	@Override
 	public void setInput(final IEditorInput input) {
-		setEditDomain(createEditDomain());
-		getEditDomain().setDefaultTool(createDefaultTool());
-		getEditDomain().setActiveTool(getEditDomain().getDefaultTool());
-		// use one "System - Wide" command stack to avoid incositensies due to
-		// undo redo
-		if (null != getSystem()) {
-			getEditDomain().setCommandStack(getSystem().getCommandStack());
+		if (!(input instanceof final IContentEditorInput contentEI)) {
+			throw new IllegalArgumentException("Diagram editors only accept ApplicationEditorInput as valid inputs!"); //$NON-NLS-1$
+		}
+
+		final IContentEditorInput currentEditorInput = (IContentEditorInput) getEditorInput();
+		if (currentEditorInput != null && currentEditorInput.getContent() != contentEI.getContent()) {
+			throw new IllegalArgumentException(
+					"Editor input with new content given to diagram editor. This is currently not supported!"); //$NON-NLS-1$
+		}
```

Comment:
Should we extract this into a separate method, so that subclasses can perform custom checks?

Reply:
Maybe later, currently there is still to much custom in the sub editors and I also would like to clean a few things in the DiagramEditors first. Then I think it will be clearer how much checks and what should be done in the sub editors. Also I have to think how I can best combine that with the ITypeEditorPage. Therefore I would prefer to move that to a later PR. But it is on my list.